typescript/radix-sdk/src/ism/routing-ism.ts

```diff
@@ -0,0 +1,180 @@
+import { AnnotatedRadixTransaction } from '../utils/types.js';
+
+import { getDomainRoutingIsmConfig } from './ism-query.js';
+
+export class RadixRoutingIsmRawReader
+  implements ArtifactReader<RawRoutingIsmArtifactConfig, DeployedIsmAddresses>
+{
+  constructor(private readonly gateway: Readonly<GatewayApiClient>) {}
+
+  async read(
+    address: string,
+  ): Promise<
+    ArtifactDeployed<RawRoutingIsmArtifactConfig, DeployedIsmAddresses>
+  > {
+    const ismConfig = await getDomainRoutingIsmConfig(this.gateway, address);
+
+    const domains: Record<number, string> = {};
+    for (const route of ismConfig.routes) {
+      domains[route.domainId] = route.ismAddress;
+    }
+
+    return {
+      artifactState: ArtifactState.DEPLOYED,
+      config: {
+        type: IsmType.ROUTING,
+        owner: ismConfig.owner,
+        domains,
+      },
+      deployed: {
+        address: ismConfig.address,
+      },
+    };
+  }
+}
+
+export class RadixRoutingIsmRawWriter
+  extends RadixRoutingIsmRawReader
+  implements ArtifactWriter<RawRoutingIsmArtifactConfig, DeployedIsmAddresses>
+{
+  constructor(
+    gateway: Readonly<GatewayApiClient>,
+    private readonly signer: RadixBaseSigner,
+    private readonly populate: RadixCorePopulate,
+    private readonly base: RadixBase,
+    private readonly accountAddress: string,
+  ) {
+    super(gateway);
+  }
+
+  async create(
+    artifact: Artifact<RawRoutingIsmArtifactConfig, DeployedIsmAddresses>,
+  ): Promise<
+    [
+      ArtifactDeployed<RawRoutingIsmArtifactConfig, DeployedIsmAddresses>,
+      TxReceipt[],
+    ]
+  > {
+    const { config } = artifact;
+
+    const routes = Object.entries(config.domains).map(
+      ([domainId, ismAddress]) => ({
+        domainId: parseInt(domainId),
+        ismAddress,
+      }),
+    );
+
+    const transactionManifest = await this.populate.createRoutingIsm({
+      from_address: this.accountAddress,
+      routes,
+    });
+
+    const receipt = await this.signer.signAndBroadcast(transactionManifest);
+    const address = await this.base.getNewComponent(receipt);
+
+    const deployedArtifact: ArtifactDeployed<
+      RawRoutingIsmArtifactConfig,
+      DeployedIsmAddresses
+    > = {
+      artifactState: ArtifactState.DEPLOYED,
+      config: artifact.config,
+      deployed: {
+        address,
+      },
+    };
+
+    return [deployedArtifact, [receipt]];
+  }
+
+  async update(
+    artifact: ArtifactDeployed<
+      RawRoutingIsmArtifactConfig,
+      DeployedIsmAddresses
+    >,
+  ): Promise<AnnotatedRadixTransaction[]> {
+    const { config, deployed } = artifact;
+    const currentConfig = await this.read(deployed.address);
+
+    const transactions: AnnotatedRadixTransaction[] = [];
+
+    // Find domains to add
+    for (const [domainId, ismAddress] of Object.entries(config.domains)) {
+      const domain = parseInt(domainId);
+      const currentIsmAddress = currentConfig.config.domains[domain];
+
```

Comment:
in the existing implementation, theres this `sleep()` thingy. do we need this still?